source/adapters/hip/enqueue.cpp

```diff
@@ -1060,7 +1060,6 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueMemImageRead(
   } catch (...) {
     return UR_RESULT_ERROR_UNKNOWN;
   }
-  return UR_RESULT_SUCCESS;
   return Result;
```

Comment:
I think the wrong `return` has been removed here, this is a behaviour change.

Reply:
This pattern is used in a lot of places here, `Result` is what gets returned. On closer inspection though I don't like it, I've updated these two entry points and I'm working on a refactor for this whole file because I think the pattern potentially drops error codes.